docs/library/forms/form.md

```diff
@@ -0,0 +1,527 @@
+            ),
+            direction="column",
+            gap="2",
+            align="stretch",
+        ),
+        name="email",
+    ),
+    on_submit=lambda _: rx.window_alert("Submitted!"),
+    reset_on_submit=True,
+)
+```
+
+In this example, the `textfield_input` has an attribute `type="email"` and the `form_message` attribute `match="typeMismatch"`. Those are required for the form to validate the input by its type. The prop `as_child="True"` is required when we use other components to construct `form_control` or `form_submit`. In the example, we have used `textfield_input` to construct the `form_control` and `button` the `form_submit`.
+
+## Form Anatomy
+
+```python eval
+rx.code_block(
+    """form_root(
+    form_field(
+        form_label(...),
+        form_control(...),
+        form_message(...),
+    ),
+    form_submit(...),
+)""",
+    language="python",
+)
+```
+
+A `form_root` contains all the parts of a form. The `form_field`, `form_submit`, etc should all be inside a `form_root`. A `form_field` can contain a `form_label`, a `form_control`, and a `form_message`. A `form_label` is a label element. A `form_control` is where the user enters the input or makes selections. By default, `form_control` is a input. You can use your own form components to construct the `form_control` and set the prop `as_child=True`.
+
+```md alert info
+The current version of Radix Forms does not support composing `form_control` with other Radix form primitives such as `checkbox`, `select`, etc.
+```
+
+The `form_message` is a validation message which is automatically wired (functionality and accessibility). When `form_control` determines the input is invalid, the `form_message` is shown. You set the `match` prop to enable [client side validation](#client-side-validation). You can set the `force_match` prop of `form_message` and `server_invalid` prop of `form_field` together to perform [server side validation](#server-side-validation).
+
+The `form_submit` is by default a button that submits the form. You can use your own button component as a `form_submit` by including that button as a child inside the `form_submit` and set the `as_child` prop to `True`.
+
+The `on_submit` prop of `form_root` accepts an event handler. It is called with the submitted form data dictionary. To clear the form after submission, set the `reset_on_submit=True` prop.
+
+## Data Submission
+
+As previously mentioned, the various pieces of data in the form are submitted together as a dictionary. The form control or the input components must have the `name` attribute. This `name` is the key to get the value from the form data dictionary. If you do not need any validation, you can include the form type components such as Checkbox, Radio Groups, TextArea directly under the `form_root`.
+
+```python demo exec
+import reflex as rx
+import reflex.components.radix.themes as rdxt
+import reflex.components.radix.primitives as rdxp
+
+class RadixFormSubmissionState(rx.State):
```

Comment:
This whole code can be simplified. We only need one `dict` var for the form data, and can render that.

Reply:
rewrote using the dict and foreach now.